app/src/main/kotlin/github/com/raccok/dota2androidapp/MainActivity.kt

```diff
@@ -26,17 +26,28 @@ import android.app.Activity
 import android.app.AlertDialog
 import android.content.SharedPreferences
 import android.os.Bundle
+import android.util.Log
 import android.widget.EditText
 import android.widget.TextView
 import android.widget.Toast
+import com.loopj.android.http.AsyncHttpClient
+import com.loopj.android.http.JsonHttpResponseHandler
+import cz.msebera.android.httpclient.Header
+import org.json.JSONObject
+import android.net.ConnectivityManager
 
 class MainActivity : Activity() {
+  private var mUserInput = ""
+  private var mHeroNames: MutableList<String> = mutableListOf()
   private var mMainTextView: TextView? = null
```

Comment:
You can use [Kotlin Android Extension](https://kotlinlang.org/docs/tutorials/android-plugin.html) to help reduce view declaration. This will also remove the `findViewById` line needed.

Reply:
Thanks, this allowed some nice refactoring. Please see if I forgot something in the according commit